coniql/caplugin.py

```diff
@@ -134,18 +168,38 @@ async def put_channels(
     async def subscribe_channel(
         self, pv: str, config: ChannelConfig
     ) -> AsyncIterator[Channel]:
-        q: asyncio.Queue[AugmentedValue] = asyncio.Queue()
-        info, meta_value = await asyncio.gather(
-            cainfo(pv), caget(pv, format=FORMAT_CTRL),
+        channel = CAChannel(pv, config)
+        # A queue that contains a monitor update and the keyword with which
+        # the channel's update_value function should be called.
+        q: asyncio.Queue[Tuple[str, AugmentedValue]] = asyncio.Queue()
+        # Monitor PV for value and alarm changes with associated timestamp.
+        value_monitor = camonitor(
+            pv, lambda v: q.put(("time_value", v)), format=FORMAT_TIME,
+        )
+        # Monitor PV only for property changes. For EPICS < 3.15 this monitor
+        # will update once on connection but will not subsequently be triggered.
+        # https://github.com/dls-controls/coniql/issues/22#issuecomment-863899258
+        meta_monitor = camonitor(
+            pv,
+            lambda v: q.put(("meta_value", v)),
+            events=DBE_PROPERTY,
+            format=FORMAT_CTRL,
         )
-        m = camonitor(pv, q.put, format=FORMAT_TIME)
         try:
-            # Hold last channel for squashing identical alarms
-            last_channel = None
+            target, first_value = await q.get()
+            yield channel.update_value(**{target: first_value})
```

Comment:
Do we want to yield the first value before doing cainfo? We'd get a temporary blip on the writeable field...

Reply:
We would. It depends whether we care about connecting more quickly or starting with complete information.

Reply:
cainfo on a connected channel should be a few milliseconds, so I'd say starting with complete information is better...

Reply:
OK